meta/saiserialize.cpp

```diff
@@ -1646,6 +1646,29 @@ std::string sai_serialize_queue_deadlock_ntf(
     return j.dump();
 }
 
+std::string sai_serialize_nat_entry(
+        _In_ const sai_nat_entry_t& nat_entry)
+{
+    SWSS_LOG_ENTER();
+
+    json j;
+
+    j["switch_id"] = sai_serialize_object_id(nat_entry.switch_id);
+    j["vr"] = sai_serialize_object_id(nat_entry.vr_id);
+    j["key.src_ip"] = sai_serialize_ipv4(nat_entry.data.key.src_ip);
```

Comment:
dont plane everything, add function that will serialize/deserialize nat_entry_data and functions that will serialize/deserialize nat_entry_mask and nat_entry_key and reuse those functions here to construct proper json object

*also* nat entry mask and key seems the same struct why not reuse the same struct ??? nat entry data struct already distinguish key and mask not need for duplicating the same struct

Reply:
@kcudnik serialize/deserialize changes done as suggested. The structures nat entry mask and key have the same fields like you said. Since they are part of the sai spec, they cannot be changed to reuse the same struct, as part of this PR.